platform/ext/target/nordic_nrf/common/native_drivers/spu.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __SPU_H__
+#define __SPU_H__
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+/**
+ * \brief SPU interrupt enabling
+ *
+ * Enable security violations outside the Cortex-M33
+ * to trigger SPU interrupts.
+ */
+void spu_enable_interrupts(void);
+
+/**
+ * \brief Reset all memory regions to being Secure
+ *
+ * Reset all (Flash or SRAM) memory regions to being Secure
+ * and have default (i.e. Read-Write-Execute allow) access policy
+ *
+ * \note region lock is not applied to allow modifying the configuration.
+ */
+void spu_regions_reset_all_secure(void);
+
+/**
+ * \brief Configre memory regions as Non-Secure
+ *
+ * Configure a range of (Flash or SRAM) memory regions as Non-Secure
+ *
+ * \note region lock is applied to prevent further modification during
+ * the current reset cycle.
+ */
+
+void spu_regions_config_non_secure( u32_t start_addr, u32_t limit_addr, bool ram);
```

Comment:
This API is a bit inconsistent. We've got a common function for configuring non-secure regions in flash and RAM but separate functions for configuring NSC regions (the one for RAM non-existing yet).
There is actually no advantage of having this `ram` parameter here as there is no common code for flash and RAM in this function. Thus, I think it would be better to have separate functions for configuring non-secure flash and RAM sections. Additionally, if one of them was not used in some fancy setup, the other one could be discarded by the linker.

Reply:
> Additionally, if one of them was not used in some fancy setup, the other one could be discarded by the linker.

This is very unlikely @anangl , I cannot think of us doing a non XIP Zephyr build on nRF platforms (that would be the only reason not to configure non-secure flash, and even this would not work if we wanted to do non-secure image updades e.g. via MCUboot...)

Reply:
> Thus, I think it would be better to have separate functions for configuring non-secure flash and RAM sections.

I will try that, if this makes the API consistent, in your opinion. The code overhead should not be a lot. (But there will be some.)

Reply:
Updated the API, to have separate functions for flash and sram NS configuration